NETProvider/src/FirebirdSql.Data.FirebirdClient/Common/DbField.cs

```diff
@@ -347,15 +347,22 @@ public void SetValue(byte[] buffer)
 						}
 						else
 						{
-							string s = this.Charset.GetString(buffer, 0, buffer.Length);
-
-							if ((this.Length % this.Charset.BytesPerCharacter) == 0 &&
-								s.Length > this.CharCount)
+							if (this.Charset.IsOctetsCharset)
 							{
-								s = s.Substring(0, this.CharCount);
+								this.Value = buffer;
```

Comment:
I see the point.
Though answer for your question in ```no```. For ```GdsStatement``` it is done here https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/NETProvider/src/FirebirdSql.Data.FirebirdClient/Client/Managed/Version10/XdrStream.cs#L403

For ```FesStatement``` value is set here by following stack trace:
```
	FirebirdSql.Data.FirebirdClient.dll!FirebirdSql.Data.Common.DbField.SetValue(byte[] buffer) Line 344
 	FirebirdSql.Data.FirebirdClient.dll!FirebirdSql.Data.Client.Common.XsqldaMarshaler.MarshalNativeToManaged(FirebirdSql.Data.Common.Charset charset, System.IntPtr pNativeData, bool fetching) Line 217
 	FirebirdSql.Data.FirebirdClient.dll!FirebirdSql.Data.Client.Native.FesStatement.Fetch() Line 439
 	FirebirdSql.Data.FirebirdClient.dll!FirebirdSql.Data.FirebirdClient.FbCommand.Fetch() Line 660
 	FirebirdSql.Data.FirebirdClient.dll!FirebirdSql.Data.FirebirdClient.FbDataReader.Read() Line 245
 	FirebirdSql.Data.UnitTests.dll!FirebirdSql.Data.UnitTests.FbCommandTests.ReadingCharOctetsTest() Line 694
```